express-notes-api/index.js

```diff
@@ -0,0 +1,112 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json')
+const notes = data.notes;
+const fs = require('fs');
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const arr = [];
+  for (const x in notes){
+    arr.push(notes[x]);
+  }
+  res.status(200).json(arr);
+})
+
+app.get('/api/notes/:id', (req, res) => {
+  const getId = req.params.id;
+  const id = Number(getId);
+  if(!Number.isInteger(id) || id <= 0){
+    const error400 = {error: "id must be a positive integer"}
+    res.status(400).json(error400);
+  }
+  else if(Number.isInteger(id) && id in notes) {
```

Comment:
you don't need to check if the id is an integer again, the check above did that.